#####################
# Octomap Generator
#####################
octomap:
    pointcloud_topic: "/semantic_pcl/semantic_pcl"
    #pointcloud_topic: "/camera/depth/points"
    #tree_type: COLOR_OCTREE 0 SEMANTICS_OCTREE_MAX 1 SEMANTICS_OCTREE_BAYESIAN 2
    tree_type: 1
    world_frame_id: "world"
    resolution: 0.1
    max_range: 5.0
    raycast_range: 3.0
    clamping_thres_min: 0.12
    clamping_thres_max: 0.97
    occupancy_thres: 0.5
    prob_hit: 0.7
    prob_miss: 0.4
    save_path : "~/map.bt"

#####################
# System Related Parameters
#####################
system/v_max: 1 #0.5 ##0.5
system/dyaw_max: 0.5
# check the relevant sdf file: depth_camera_new.sdf
system/camera/pitch: [0.0] #sensor_pitch: 0.0
system/camera/horizontal: [60.0] #sensor_hor_fov: 60
system/camera/vertical: [45.0] #sensor_ver_fov: 45
system/bbx/x: 0.2
system/bbx/y: 0.2
system/bbx/z: 0.2
system/bbx/overshoot: 0.5
system/localization/use_gazebo_ground_truth: true

# Gain Computation Parameters
nbvp/gain/free: 0.0
nbvp/gain/occupied: 0.0
nbvp/gain/unmapped: 1.0
nbvp/gain/area: 0.0
nbvp/gain/probabilistic: 0.0
nbvp/gain/range: 5.0 # Maximum distance of volumes to be considered for the gain computation (double, default: 1.0)
nbvp/gain/zero: 0.0
nbvp/gain/degressive_coeff: 0.5

inspection_throttle: 0.5
pcl_throttle: 0.33 # default 0.33
output/file/name: "gains.csv"

# Path Planning Parameters: rrt_planner.cpp
nbvp/tree/extension_range: 1
nbvp/tree/initial_iterations: 10
nbvp/tree/cuttoff_iterations: 300
nbvp/tree/exact_root: false
nbvp/dt: 0.1

# Exploration Loop: exploration_planner_node.cpp
exploration/log/on: true
exploration/dt: 2.0
exploration/num_iterations: 500

# Logging Parameters
nbvp/log/throttle: 0.25 # ?? 
nbvp/log/on: true

# Exploration Area 
bbx/minX:  -6.2 #-10.0
bbx/maxX: 3.0 #10.0
bbx/minY: -2.2 #-10.0
bbx/maxY: 6.6 #10.0
bbx/minZ: -0.1
bbx/maxZ: 2.5 #3.0
bbx/softBounds: false

############
## Octomap Manager settings
############
tf_frame: "world"
robot_frame: "base_point_cloud"
resolution: 0.15
mesh_resolution: 1.0
visualize_max_z: 5.0
sensor_max_range: 8.0
map_publish_frequency: 0.2
probability_hit: 0.7
probability_miss: 0.4
threshold_min: 0.12
threshold_max: 0.97
threshold_occupancy: 0.7
treat_unknown_as_occupied: false
latch_topics: false
max_free_space: 0
min_height_free_space: 0
change_detection_enabled: false
objects_of_interest: ["book", "person","desk","monitor","cabinet","sofa","vase"]
confidence_threshold: 0.7 
num_of_visits_threshold: 3 

# Planning Methods
# Volumetric 0
# Rear side voxel  1 
# Semantic rear side voxel  2 
# rear side entropy 3 
# Semantic rear side entropy 4 
# Pure entropy 5
# Avg entropy 6 
# occlusion aware 7 
# unobserved voxel 8 
# semantic Visible voxels with confidance 9 (proposed 1) 
# semantic occlusion aware 10 (proposed 2 - The planner crashes)   
# semantic obj of Interest depending on number of visits 11 (proposed 3) 

/utility/method: 0
/output/file/name: "1.csv"
